lib/data_update_scripts/20210423155327_migrate_relevant_fields_from_users_to_users_settings.rb

```diff
@@ -1,86 +1,9 @@
-                    WHEN config_theme='night_theme' THEN 2
-                    WHEN config_theme='pink_theme' THEN 3
-                    WHEN config_theme='ten_x_hacker_theme' THEN 4
-                    ELSE 0
-                END
-                config_theme,
-                COALESCE(display_announcements, true),
-                COALESCE((data ->> 'display_email_on_profile')::boolean, false) as display_email_on_profile,
-                COALESCE(display_sponsors, true),
-                CASE WHEN editor_version='v2' THEN 0
-                      WHEN editor_version='v1' THEN 1
-                      ELSE 0
-                END
-                editor_version,
-                experience_level,
-                COALESCE(feed_mark_canonical, false),
-                COALESCE(feed_referential_link, true),
-                feed_url,
-                inbox_guidelines,
-                CASE WHEN inbox_type='private' THEN 0
-                      WHEN inbox_type='open' THEN 1
-                      ELSE 0
-                END
-                inbox_type,
-                COALESCE(permit_adjacent_sponsors, true),
-                users.id AS user_id,
-                NOW(),
-                NOW()
-              FROM users
-              JOIN profiles
-                ON profiles.user_id = users.id
-            )
-            INSERT INTO users_settings (brand_color1, brand_color2, config_font, config_navbar, config_theme, display_announcements, display_email_on_profile, display_sponsors, editor_version, experience_level, feed_mark_canonical, feed_referential_link, feed_url, inbox_guidelines, inbox_type, permit_adjacent_sponsors, user_id, created_at, updated_at)
-              SELECT * FROM settings_data
-              ON CONFLICT (user_id) DO UPDATE
-                SET brand_color1 = EXCLUDED.brand_color1,
-                    brand_color2 = EXCLUDED.brand_color2,
-                    config_font = EXCLUDED.config_font,
-                    config_navbar = EXCLUDED.config_navbar,
-                    config_theme = EXCLUDED.config_theme,
-                    display_announcements = EXCLUDED.display_announcements,
-                    display_email_on_profile = EXCLUDED.display_email_on_profile,
-                    display_sponsors = EXCLUDED.display_sponsors,
-                    editor_version = EXCLUDED.editor_version,
-                    experience_level = EXCLUDED.experience_level,
-                    feed_mark_canonical = EXCLUDED.feed_mark_canonical,
-                    feed_referential_link = EXCLUDED.feed_referential_link,
-                    feed_url = EXCLUDED.feed_url,
-                    inbox_guidelines = EXCLUDED.inbox_guidelines,
-                    inbox_type = EXCLUDED.inbox_type,
-                    permit_adjacent_sponsors = EXCLUDED.permit_adjacent_sponsors,
-                    updated_at = NOW();
-          SQL
-        )
+      User.ids.each do |id|
+        MigrateUserSettingsWorker.perform_async id
```

Comment:
We can do 600,000 queue pushes in the time limit? If I understand this  - we're pulling all ids from the db, and enqueuing one job per user id, and waiting for the queue to drain in the background so that this job can complete successfully.

Reply:
Good point. Every `perform_async` is a Redis call. 620k calls at 1ms each (arbitrarary number :D) is 10.3 minutes.
What's the timeout on workers? (this file is execute inside a worker, `DataUpdateWorker`)

Reply:
> What's the timeout on workers? (this file is execute inside a worker, `DataUpdateWorker`)

@rhymes I don't think we have a time limit on workers. The [`Feeds::ImportArticlesWorker` exceeds 10 minutes at least 10% of the time](https://app.datadoghq.com/apm/resource/sidekiq/sidekiq.job/2588a0e64f978d2c?end=1621955565278&env=production&index=apm-search&paused=false&query=env%3Aproduction%20service%3Asidekiq%20operation_name%3Asidekiq.job%20resource_name%3A%22Feeds%3A%3AImportArticlesWorker%22&start=1621350765278):

<img width="118" alt="image" src="https://user-images.githubusercontent.com/108205/119522761-3c30e900-bd4a-11eb-9a5c-2c271e752d67.png">